OpenRA.Mods.Common/Lint/CheckTranslationReference.cs

```diff
@@ -52,20 +47,55 @@ void ILintPass.Run(Action<string> emitError, Action<string> emitWarning, ModData
 							if (referencedKeys.Contains(key))
 								continue;
 
-							if (!translation.HasMessage(key))
-								emitError($"{key} not present in {language} translation.");
-
+							testKey(key);
 							referencedKeys.Add(key);
 						}
 					}
 				}
 			}
+		}
+
+		void ILintMapPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Map map)
+		{
+			if (map.TranslationDefinitions == null)
+				return;
+
+			// TODO: Check all available languages.
+			var language = "en";
+			var modTranslation = new Translation(language, modData.Manifest.Translations, modData.DefaultFileSystem);
+			var mapTranslation = new Translation(language, FieldLoader.GetValue<string[]>("value", map.TranslationDefinitions.Value), map);
+
+			TestTraits(map.Rules, (key) =>
+				{
```

Comment:
The indentation seems off by one level here.

Reply:
There is also unnecessary parenthesis around `key` here and below.

Reply:
I think the indentation is correct here. We always put a tab when we move one line code to a new line

Reply:
We don't when there is a new scope.

Reply:
weird but ok